cfme/common/host_views.py

```diff
@@ -0,0 +1,365 @@
+# -*- coding: utf-8 -*-
+from widgetastic.widget import ParametrizedView, Text, View
+from widgetastic_patternfly import (
+    BootstrapSelect,
+    CheckableBootstrapTreeview,
+    Dropdown,
+    FlashMessages,
+    Tab
+)
+from widgetastic.utils import ParametrizedLocator
+
+from cfme.base.login import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound, ManyItemsFound
+from widgetastic_manageiq import (
+    BaseItem,
+    BaseListItem,
+    BaseQuadIconItem,
+    BaseTileIconItem,
+    BootstrapTreeview,
+    BreadCrumb,
+    Button,
+    Checkbox,
+    Input,
+    ItemsToolBarViewSelector,
+    PaginationPane,
+    Search,
+    SummaryTable,
+    Table,
+    TimelinesView
+)
+
+
+class ComputeInfrastructureHostsView(BaseLoggedInPage):
+    """Common parts for host views."""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+    flash = FlashMessages(
+        './/div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+        'contains(@class, "flash_text_div")]'
+    )
+
+    @property
+    def in_compute_infrastructure_hosts(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ["Compute", "Infrastructure", "Hosts"]
+        )
+
+
+class HostQuadIconItem(BaseQuadIconItem):
+
+    @property
+    def no_vm(self):
+        return int(self.browser.text(self.QUADRANT.format(pos="a")))
+
+    @property
+    def status(self):
+        return self.browser.get_attribute("style", self.QUADRANT.format(pos="b"))
+
+    @property
+    def vendor(self):
+        return self.browser.get_attribute("alt", self.QUADRANT.format(pos="c"))
+
+    @property
+    def creds(self):
+        return self.browser.get_attribute("alt", self.QUADRANT.format(pos="d"))
+
+
+class HostTileIconItem(BaseTileIconItem):
+    quad_icon = ParametrizedView.nested(HostQuadIconItem)
+
+
+class HostListItem(BaseListItem):
+    pass
+
+
+class HostItem(BaseItem):
+    quad_item = HostQuadIconItem
+    list_item = HostListItem
+    tile_item = HostTileIconItem
+
+
+class HostDetailsToolbar(View):
+    """Represents host toolbar and its controls."""
+    monitoring = Dropdown(text="Monitoring")
+    configuration = Dropdown(text="Configuration")
+    policy = Dropdown(text="Policy")
+    power = Dropdown(text="Power")
+
+
+class HostDetailsSummaryView(View):
```

Comment:
**Required**

I'd like to keep these class names consistent, we've been using 'Entities' for this section of the page in other modules. So `HostDetailsEntities` here, and then in `HostDetailsView` below there should be an `entities` attribute instead of 'contents'

Reason being so that we can iterate over a set of views and consistently validate against a given attribute (toolbar, sidebar, entities, paginator)